modules/dxp/apps/portal-search-tuning/portal-search-tuning-rankings-web/src/main/java/com/liferay/portal/search/tuning/rankings/web/internal/request/SearchRankingRequest.java

```diff
@@ -95,6 +97,55 @@ private String _getOrderByType() {
 			"search-ranking-order-by-type", "asc");
 	}
 
+	private BooleanQuery _getQuery() {
+		BooleanQuery booleanQuery = _queries.booleanQuery();
+		String keywords = _searchContext.getKeywords();
+		String scope = GetterUtil.getString(
+			_httpServletRequest.getParameter("scope"), "all");
+		String status = GetterUtil.getString(
+			_httpServletRequest.getParameter("status"), "all");
+
+		if (!Validator.isBlank(keywords)) {
+			booleanQuery.addMustQueryClauses(
+				_queries.booleanQuery(
+				).addShouldQueryClauses(
+					_queries.term(RankingFields.NAME, keywords),
+					_queries.term(RankingFields.ALIASES, keywords)
+				));
+		}
+
+		if (!Objects.equals(scope, "all")) {
+			if (Objects.equals(scope, "blueprint")) {
+				booleanQuery.addMustNotQueryClauses(
+					_queries.term(
+						RankingFields.SXP_BLUEPRINT_EXTERNAL_REFERENCE_CODE,
+						StringPool.BLANK));
+			}
+			else if (Objects.equals(scope, "site")) {
```

Comment:
Hi @AlmirFe 
I'm pretty sure creating a queries.term() with StringPool.BLANK isn't what we want to be doing. Did you see this elsewhere in the code, and is there a way to test the scope through the UI?

Reply:
Instead of MUST_NOT if the field is BLANK. Use an Exists query like [here](https://github.com/liferay/liferay-portal/blob/6ce0b9eb049e99a50285a324a3bd4afd76d340ab/modules/dxp/apps/portal-workflow/portal-workflow-metrics-rest-impl/src/main/java/com/liferay/portal/workflow/metrics/rest/internal/resource/v1_0/NodeMetricResourceImpl.java#L255).

Reply:
Hi @joshuacords, This is a way I found to make elasticsearch only return results where the field is empty. Basically, result rankings do not have the scope field. So we need to check if the SXP_BLUEPRINT_EXTERNAL_REFERENCE_CODE field is filled in to know if the scope is SXPBlueprint and do the same with the GROUP_EXTERNAL_REFERENCE_CODE field. In this case, I want to make sure the scope is NOT SXPBlueprint. Then I added StringPool.BLANK to the value. But we can talk about that to see if there is a better way to do that.